json encode double fixed array

### Describe the bug

Code: https://play.vlang.io/p/cae76f07e6

```v
import json

struct Base {
	options  Options
	profiles Profiles
}

struct Options {
	cfg [6]u8
}

struct Profiles {
	cfg [4][7]u8 // <- cause of C error
}

fn main() {
	println(Base{}) // OK
	println(json.encode(Base{})) // C error
}
```

### Reproduction Steps

Run the code at the playground

### Expected Behavior

Create json string from the object. Regular print is 
```
Base{
    options: Options{
        cfg: [0, 0, 0, 0, 0, 0]
    }
    profiles: Profiles{
        cfg: [[0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0]]
    }
}
```

### Current Behavior

Output:

```
/tmp/v_60000/code.01J923657XT1ASJX52P6NTVDTT.tmp.c:2610: error: '{' expected (got ";")
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
Exited with error status 1
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.8 aa3ab3f

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.8 aa3ab3f
OS: linux, Debian GNU/Linux 12 (bookworm) (VM)
Processor: 2 cpus, 64bit, little endian, Intel(R) Xeon(R) CPU E5-2680 v4 @ 2.40GHz

getwd: /home/admin/playground
vexe: /home/admin/v/v
vexe mtime: 2024-09-30 18:25:27

vroot: OK, value: /home/admin/v
VMODULES: OK, value: .vmodules
VTMP: OK, value: /tmp/v_0

Git version: git version 2.39.5
Git vroot status: Error: fatal: detected dubious ownership in repository at '/home/admin/v'
To add an exception for this directory, call:

	git config --global --add safe.directory /home/admin/v
.git/config present: true

CC version: cc (Debian 12.2.0-14) 12.2.0
thirdparty/tcc status: Error: fatal: detected dubious ownership in repository at '/home/admin/v/thirdparty/tcc'
To add an exception for this directory, call:

	git config --global --add safe.directory /home/admin/v/thirdparty/tcc
 Error: fatal: detected dubious ownership in repository at '/home/admin/v/thirdparty/tcc'
To add an exception for this directory, call:

	git config --global --add safe.directory /home/admin/v/thirdparty/tcc

```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/json/json_fixed_array_test.v
+++ b/vlib/json/json_fixed_array_test.v
@@ -1,0 +1,25 @@
+import json
+
+struct Base {
+	options  Options
+	profiles Profiles
+}
+
+struct Options {
+	cfg [6]u8
+}
+
+struct Profiles {
+	cfg [4][7]u8
+}
+
+fn test_main() {
+	a := json.encode(Base{})
+	println(a)
+	assert a.contains('"cfg":[[0,0,0,0,0,0,0],[0,0,0,0,0,0,0],[0,0,0,0,0,0,0],[0,0,0,0,0,0,0]]')
+
+	b := json.decode(Base, a)!
+	assert b.options.cfg.len == 6
+	assert b.profiles.cfg.len == 4
+	assert b.profiles.cfg[0].len == 7
+}

--- a/vlib/v/gen/c/json.v
+++ b/vlib/v/gen/c/json.v
@@ -983,6 +983,7 @@ fn (mut g Gen) decode_array(utyp ast.Type, value_type ast.Type, fixed_array_size
 	mut fixed_array_idx := ''
 	mut fixed_array_idx_increment := ''
 	mut array_element_assign := ''
+	is_array_fixed_val := g.table.final_sym(value_type).kind == .array_fixed
 	if utyp.has_flag(.option) {
 		if fixed_array_size > -1 {
 			fixed_array_idx += 'int fixed_array_idx = 0;'
@@ -994,7 +995,11 @@ fn (mut g Gen) decode_array(utyp ast.Type, value_type ast.Type, fixed_array_size
 			array_free_str += 'array_free(&res.data);'
 		}
 	} else {
-		if fixed_array_size > -1 {
+		if is_array_fixed_val {
+			fixed_array_idx += 'int fixed_array_idx = 0;'
+			array_element_assign += 'memcpy(res[fixed_array_idx], val, sizeof(${styp}));'
+			fixed_array_idx_increment += 'fixed_array_idx++;'
+		} else if fixed_array_size > -1 {
 			fixed_array_idx += 'int fixed_array_idx = 0;'
 			array_element_assign += 'res[fixed_array_idx] = val;'
 			fixed_array_idx_increment += 'fixed_array_idx++;'
@@ -1008,6 +1013,15 @@ fn (mut g Gen) decode_array(utyp ast.Type, value_type ast.Type, fixed_array_size
 	mut s := ''
 	if is_js_prim(styp) {
 		s = '${styp} val = ${fn_name}((cJSON *)jsval); '
+	} else if is_array_fixed_val {
+		s = '
+		${result_name}_${styp} val2 = ${fn_name} ((cJSON *)jsval);
+		if(val2.is_error) {
+			${array_free_str}
+			return *(${result_name}_${ret_styp}*)&val2;
+		}
+		${styp} val;
+		memcpy(&val, (${styp}*)val2.data, sizeof(${styp}));'
 	} else {
 		s = '
 		${result_name}_${styp} val2 = ${fn_name} ((cJSON *)jsval);

